internal/js-dependency-scanner/src/main/java/org/wso2/security/tools/scanner/dependency/js/preprocessor/ResourceDownloader.java

```diff
@@ -0,0 +1,141 @@
+
+import java.io.File;
+import java.io.IOException;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.time.LocalDateTime;
+import java.util.Date;
+import java.util.List;
+import java.util.Locale;
+import java.util.concurrent.TimeUnit;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Abstract class for resource downloader. This has common methods which is needed to be check to download
+ * relevant weekly release of products.
+ */
+public abstract class ResourceDownloader {
+
+    private static final Logger log = Logger.getLogger(ResourceDownloader.class);
+
+    /**
+     * This method calculate and returns the no of days between published date of particular weekly release
+     * and scan date (Current system date).
+     *
+     * @param releaseDate published date of particulae weekly release.
+     * @return no of days between published date of particular weekly release
+     * and scan date (Current system date)
+     * @throws ParseException Exception occurred parsing the string to date format.
+     */
+    long getDateDiffFromLastWeeklyRelease(String releaseDate)
+            throws ParseException {
+
+        SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd", Locale.ENGLISH);
+        LocalDateTime now = LocalDateTime.now();
+        Date firstDate = sdf.parse(releaseDate);
+        Date secondDate = sdf.parse(now.toString());
+
+        long diffInMillis = Math.abs(secondDate.getTime() - firstDate.getTime());
+        return TimeUnit.DAYS.convert(diffInMillis, TimeUnit.MILLISECONDS);
+    }
+
+    /**
+     * Check whether the given asset name is weekly release or not. It checks with the regular expression
+     * which indicates the version pattern. (It checks whether the asset contains -x.x.x-).
+     *
+     * @param name Asset name
+     * @return True if it is weekly release, False if it is not a weekly release.
+     */
+    boolean isWeeklyRelease(String name) {
+        boolean isWeeklyRelease = false;
+        Pattern pattern = Pattern.compile(JSScannerConstants.VERSION_REGEX);
+        Matcher matcher = pattern.matcher(name);
+        if (matcher.find()) {
+            isWeeklyRelease = true;
+        }
+        return isWeeklyRelease;
+    }
+
+    /**
+     * Check whether the given asset name is GA release or not. It checks with the regular expression
+     * which indicates the version pattern. (It checks whether the asset contains -x.x.x.).
+     *
+     * @param name asset name
+     * @return True if it is weekly release, False if it is not a weekly release.
+     */
+    boolean isGARelease(String name) {
+        boolean isGARelease = false;
+        Pattern pattern = Pattern.compile((JSScannerConstants.GA_RELEASE_VERSION_REGEX));
+        Matcher matcher = pattern.matcher(name);
+        if (matcher.find()) {
+            isGARelease = true;
+        }
+        return isGARelease;
+    }
+
+    /**
+     * Create Directory
```

Comment:
"Create Directory" - for what purpose? Be more descriptive with JavaDoc comments